### PR TITLE
Added shortcake UI plugin integration

### PR DESCRIPTION
**Pull request by dashaluna**
#1

### CODE REVIEW

File: includes/class-content-toc.php [hunk 1]
```diff
@@ -22,6 +22,9 @@ protected function __construct() {
 
 		// Register shortcode
 		add_shortcode( 'hm_content_toc', array( $this, 'shortcode' ) );
+
+		// Setup shortcake UI integration
+		require_once( __DIR__ . '/shortcake-integration.php' );
```
Comment (dashaluna) [inline includes/class-content-toc.php]: Is this a good approach to include 3rd party plugins integrations?

**Comment by dashaluna** [comment]
I've noticed that if I specify element as `<h3>` i.e. with closing and opening tags in shortcake UI, then I get the following error in logs: 
`Unknown modifier 'p' in /hm-content-toc/includes/class-content-toc.php on line 151`. 

I wonder if it adds `p` autowrap and my regex isn't coping with it. Not sure how to tackle this one. Any suggestions, @tcrsavage :)

**P.S.** this is might be due to the fact that other PR that deal with escaping regex isn't merge yet. Ignore this and I'll double check this again after another PR is merged :)

**Reference:** Shortcake fixed it for 4.1, but it got broken again from 4.2 as WP made changes to MCE Views. https://github.com/fusioneng/Shortcake/pull/179

**Update:** Gonna leave this for now.. elements specified with `<>` will be regex escaped and won't match anything and will be skipped. Nothing breaks, just doesn't match them.

**Comment by tcrsavage** [comment]
@dashaluna looks fine to me. May want to consider an admin notice about the shortcake requirement if it's not installed.

**Comment by dashaluna** [comment]
@tcrsavage Hm, shortcake UI isn't a requirement really. It just does integration if that plugin is already present, otherwise it does nothing. I guess I can display a note saying if they know about shortcake and want to check it out... What do you think?

**Pull request merged**